src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -426,6 +432,26 @@ public String actualPathForPublicPath(String publicPath) {
         return null;
     }
 
+    /**
+     * Checks whether a path (defined by a URI) contains a certain sub-path (defined by a URI).
+     *
+     * @param path    URI to be checked
+     * @param subPath URI to check
+     * @throws IllegalArgumentException if the provided path does not contain the provided sub-path
+     */
+    public static void sanitizeByCheckingIfPathContainsSubPathElseThrow(URI path, URI subPath) {
```

Comment:
Two things on top:
1. We have to make sure here, though, that this works out well. Cause previously we allowed files to still be in the temporary path, not the expected one. So if now a file was not yet moved it worked previously, it doesn't work now. But it's quite possible that it wasn't even necessary before. But we need to make sure.
2. Do we really want these null checks? Shouldn't we expect both URIs to be non-null? Add corresponding annotations? For the current implementation, it works out but if later the path is set to null for some reason, the subpath has a free pass.